.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and test
+
+on: ["push"]
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+
+      - name: Setup Node.js 14.x
+        uses: actions/setup-node@master
+        with:
+          node-version: 14.x
+
+      - name: Install Dependencies
+        run: yarn --frozen-lockfile --ignore-scripts
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Build check
+        run: yarn lerna run build:check
+
+      - name: Build packages
+        run: yarn lerna run build:test
+
+      - name: Test
+        run: yarn test -i
+
+      - name: Test size
+        run: yarn lerna run test:size
+
+      - name: Test flow
+        run: yarn mobx test:flow
+
+      - name: Test performance
+        run: yarn mobx test:performance
```

Comment:
Initially I wanted to have multiple jobs (like build, test, test size) the same way it was implemented in CircleCI config. On CircleCI it was done using [persist_to_workspace](https://circleci.com/docs/configuration-reference/) option:

https://github.com/mobxjs/mobx/blob/42f8ac057ec70c508232339016cc7249123f0fd0/.circleci/config.yml#L25

There is an alternative way doing so using GitLab actions: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow

But each time I added `actions/upload-artifact@v3` to the config the pipeline mysteriously disappeared from the GitHub pipeline list and I saw only outdated CircleCI: https://github.com/mobxjs/mobx/pull/3604/commits/b9b473df637212b5553cd540086bce66fd202e90

<img width="536" alt="Screenshot 2023-01-10 at 11 55 32" src="https://user-images.githubusercontent.com/22447849/211505749-f6ad9d8b-de8b-476d-b977-28f02fe44eee.png">

So I decided to keep it flat and it worked out.